djproxy/views.py

```diff
@@ -135,6 +135,10 @@ def proxy(self):
         headers['X-Forwarded-Host'] = self.request.get_host()
         headers['X-Forwarded-For'] = self.xff
 
+        for prefix, url in self.reverse_urls:
+            if url == self.proxy_url and self.request.path.startswith(prefix):
```

Comment:
I think it should be `self.proxy_url.startswith(url)`, not `url == self.proxy_url`:

Consider the proxy generated by the following call to `generate_routes`:

```python
generate_routes({
    'service_one': {
        'base_url': 'https://yahoo.com/',
        'prefix': '/yahoo/'
    }
})
```

If I go to `/yahoo/foo/bar`, then `proxy_url` will be https://yahoo.com/foo/bar/, and won't match `https://yahoo.com/`.

The definition for proxy_url: https://github.com/yola/djproxy/blob/x_forwarded_prefix/djproxy/views.py#L30-L32

That attribute is used primarily to calculate what URL to hit when the request is sent upstream.